geoips/plugin_registry.py

```diff
@@ -19,6 +19,121 @@
 LOG = logging.getLogger(__name__)
 
 
+class PluginRegistry:
+    """Plugin Registry class definition.
+
+    Represents all of the plugins found in all of the available GeoIPS packages.
+    This class will load a plugin when requested, rather than loading all plugins when
+    GeoIPS is instantiated.
+    """
+    def __init__(self, _test_registry_files=[]):
+        # Use this for unit testing
+        if _test_registry_files:
+            self.registry_files = _test_registry_files
+        # Use this for normal operation and collect the registry files
+        else:
+            self.registry_files = []  # Collect the paths to the registry files here
+            for pkg in get_entry_point_group("geoips.plugin_packages"):
+                self.registry_files.append(
+                    str(resources.files(pkg.value) / "registered_plugins.yaml")
+                )
+
+    @property
+    def registered_plugins(self):
+        """Find all plugins in registered plugin packages.
+
+        Search the ``registered_plugins.yaml`` of each registered plugin package and add
+        them to the _registered_plugins dictionary
+        """
+        # Load the registries here and return them as a dictionary
+        if not hasattr(self, "_registered_plugins"):
+            self._registered_plugins = {}
+            self._interface_mapping = {}
```

Comment:
Is it okay that `self._interface_mapping` isn't set unless `self.registered_plugins` has been called? Should there be a separate property for `self._interface_mapping`? Something like:

```
def set_property_attrs(self):
    # Do what is currently in `registered_plugins`
    # I'm not sure if I like the name `set_property_attrs` so choose whatever you think is appropriate
    # Maybe this function should start with an underscore? I'm not sure.

@property
def registered_plugins(self):
    if not hasattr(self, "_registered_plugins"):
        self.set_property_attrs()
    return self._registered_plugins

@property
def interface_mapping(self):
    if not hasattr(self, "_interface_mapping"):
        self.set_property_attrs()
    return self._interface_mapping

Reply:
I'll do this now! And yes, I think this should be a private method.